src/main/java/com/newrelic/telemetry/opentelemetry/export/MetricPointAdapter.java

```diff
@@ -0,0 +1,179 @@
+      return buildDoublePointMetrics(descriptor, type, attributes, (DoublePoint) point);
+    }
+    if (point instanceof SummaryPoint) {
+      return buildSummaryPointMetrics(descriptor, attributes, (SummaryPoint) point);
+    }
+    return emptyList();
+  }
+
+  private Collection<Metric> buildMetricsFromSimpleType(
+      Descriptor descriptor,
+      Type type,
+      Attributes attributes,
+      double value,
+      long epochNanos,
+      long startEpochNanos) {
+    switch (type) {
+      case NON_MONOTONIC_LONG:
+      case NON_MONOTONIC_DOUBLE:
+        return singleton(
+            new Gauge(descriptor.getName(), value, NANOSECONDS.toMillis(epochNanos), attributes));
+
+      case MONOTONIC_LONG:
+      case MONOTONIC_DOUBLE:
+        return singleton(
+            new Count(
+                descriptor.getName(),
+                value,
+                NANOSECONDS.toMillis(startEpochNanos),
+                NANOSECONDS.toMillis(epochNanos),
+                attributes));
+
+      default:
+        // maybe log something about unhandled types?
+        break;
+    }
+    return emptyList();
+  }
+
+  private boolean isNonMonotonic(Type type) {
+    return type != Type.NON_MONOTONIC_DOUBLE && type != Type.NON_MONOTONIC_LONG;
+  }
+
+  private Collection<Metric> buildDoublePointMetrics(
+      Descriptor descriptor, Type type, Attributes attributes, DoublePoint point) {
+
+    double value = point.getValue();
+    if (isNonMonotonic(type)) {
+      DeltaDoubleCounter deltaDoubleCounter =
+          deltaDoubleCountersByDescriptor.computeIfAbsent(
+              new Key(descriptor, type, point.getLabels()), d -> new DeltaDoubleCounter());
+      value = deltaDoubleCounter.delta(point);
+    }
+    return buildMetricsFromSimpleType(
+        descriptor, type, attributes, value, point.getEpochNanos(), timeTracker.getPreviousTime());
+  }
+
+  private Collection<Metric> buildLongPointMetrics(
+      Descriptor descriptor, Type type, Attributes attributes, LongPoint point) {
+    long value = point.getValue();
+    if (isNonMonotonic(type)) {
+      DeltaLongCounter deltaLongCounter =
+          deltaLongCountersByDescriptor.computeIfAbsent(
+              new Key(descriptor, type, point.getLabels()), d -> new DeltaLongCounter());
+      value = deltaLongCounter.delta(point);
+    }
+    return buildMetricsFromSimpleType(
+        descriptor, type, attributes, value, point.getEpochNanos(), timeTracker.getPreviousTime());
+  }
+
+  private Collection<Metric> buildSummaryPointMetrics(
+      Descriptor descriptor, Attributes attributes, SummaryPoint point) {
+    List<ValueAtPercentile> percentileValues = point.getPercentileValues();
+
+    double min = Double.NaN;
+    double max = Double.NaN;
+    for (ValueAtPercentile percentileValue : percentileValues) {
+      if (percentileValue.getPercentile() == 0.0) {
+        min = percentileValue.getValue();
+      }
+      if (percentileValue.getPercentile() == 100.0) {
```

Comment:
This appears to make some assumptions about the data inside the list of `ValueAtPercentile`.  Are we guaranteed to have both extremes (0% and 100%), or would it be possible for the max to actually be at 97% or similar?

Reply:
the way the aggregation is written uses 100.0 & 0.0 as the slots for max and min. I'll add a comment.

Reply:
comment added. please take a look